### ISSUE TITLE
GitBook and Every Page is Page One

### ISSUE THREAD

**Issue by jaredmorgs** [GitbookIO/feedback#31]
I've been having a productive discussion on Write The Docs Slack about GitBook and how it fits into the emerging docs trend of Every Page is Page One (EPPO)

http://everypageispageone.com/

Some of the feedback I've received to date from other professional tech writers seems to echo the following sentiment:

> My biggest issue with gitbook is that it is designed to create traditional linear-format books, and that’s not really a format I work in much any more for docs. 

Another perspective about EPPO and GitBook is revealed in the following long post by David Ryan:

> And I echo the above about GitBook, except for @jaredmorgs praise for the local install. IMHO that was a mistake. We were hoping that they would focus on filling the gaps for “private team documentation tool with version control” that even Git Pages leaves wide open, but curiously, not so.
https://writethedocs.slack.com/archives/general/p1465777750000002

I attempted to re-frame the discussion by saying the following:

> I feel that GitBook serves a need for small projects looking for a way to host documentation while at the same time allowing for collaboration and updating ease. 
I agree it would not scale when you are looking at EPPO or other more modern ways of presenting info to users. 
One think I quite like about GitBook is that it has a FAQ template that allows small projects to create a quick help portal for their users.
So you get docs, and basic user help in one platform.
That allows you to remove the often ubiquitous FAQ sections buried in guides and single source them outside a guide. 
That being said, GitBook is open source… So if you don’t see the features you want, open an issue and pitch your use-case. https://writethedocs.slack.com/archives/general/p1465778756000006

Which David Ryan then clarified with the following statement:

> to my understanding, it’s built “on” open source, but sure - raise a req if you’re so inclined, but gitbook is git “book”, which is the fundamental problem from my side and why I had to build my own solution in the first place.
as the thread above tackles, every page is now page one, so making standalone software to make books is a curious choice. don’t get me wrong, it’s a great service and they seem to patch their standalone bugs pretty quickly. but i can’t use if for the docs we need on the project I’m doing here in the US right now, for example 
Dropbox Paper is almost getting there, but making the same mistake Evernote and Google make in ignoring the versioning and fundamental issues of re-use.https://writethedocs.slack.com/archives/general/p1465779173000012

I wonder if GitBook could evolve into allowing more of an EPPO model for docs. I'm not suggesting that it would be easy to adapt GitBook away from the traditional book model, but it might be worth considering.

I will say that I'm very, very impressed with the editing environment and ease of book creation you have created for writers like me. I only hope you can continue to innovate GitBook into a platform that can be used by progressive Tech Writers who are desperate for tooling that allows them to cater for Web 2.0

**Comment by AaronO** [comment]
@jaredmorgs Thanks for the great feedback.

This is actually the direction we're pushing in, a lot of the recent work done on GitBook v3, the new editor and upcoming themes have precisely the documentation use case in mind.

Traditional book publishing is not our main focus, we in fact axed all book publishing specific features early this year (covers, selling, kindle/amazon publishing).

Here's a few of the features we've been working on and their progress (we haven't announced any of them yet):
- [x] GitBook toolchain v3 
    - [x] Better theming support
        - [x] Make it easier to build themes
        - [x] Make it easy for users to do small html tweaks (e.g: header, footer) without having to build a whole new theme
    - [x] Dev & open source friendly
         - [x] Support docs along side code (GitBook docs no longer have to be their own repo)
    - [x] Lots of small things, like:
        - [x] Anchor support (many TOC entries for one file)
- [~] GitBook Editor (v6 in alpha at the moment)
    - [x] Fullblown WYSIWYG (loads and saves to markdown)
        - [x] Nice table editing
        - [ ] Accessible to less technical users but is still backed by markdown+git
    - [ ] Better plugin support
        - [ ] Official plugins get first class integration with optimized GUIs (instead of "arcane" templating syntax)
        - [ ] All plugins are easier to add and configure directly from the editor
    - [ ] Better team workflow tools
        - [ ] "Pull requests"
        - [ ] Inline comments accessible from editor
    - [~] Misc improvements
        - [~] Stability improvements
        - [x] Performance improvements
- [ ] GitBook.com
    - [ ] Better onboarding
        - [ ] More relevant project templates (see themes below)
        - [ ] Team / collaboration onboarding
    - [ ] Better UX
- [x] New official themes and plugins (better suited to common use cases we want to support)
    - [x] API Docs
    - [x] FAQ
    - [ ] More to come and tweaks to above depending on use feedback

What would be awesome is if you could provide feedback, on features, small or big, that you think would make GitBook awesome for tech writers, that would be extremely helpful !

**Comment by jaredmorgs** [comment]
Great feature road map here, Aaron.

One thing that attracted me to GitBook was its support for AsciiDoc. But what I've found is that the default book template is markdown. 

I'm finding that I manually change each default file to AsciiDoc, and I also have to change the SUMMARY.adoc link syntax.

It would be great as part of book creation to let users select between Markdown and AsciiDoc in the UI.

**Comment by AaronO** [comment]
@jaredmorgs Good idea, might make sense to have an AsciiDoc project template for tech docs.

One downside of AsciiDoc though is that it's harder to parse that markdown, and there's no official JS implementation (besides a cross-compiled version via `opal.js`), so it might be hard for us to provide WYSIWYG features for AsciiDoc.

Would love to hear more thoughts from you or fellow tech writers on features you think would make GitBook great.

BTW, any idea on markdown/asciidoc split in your circles ?

**Comment by jaredmorgs** [comment]
@AaronO I'd be happy to collaborate with you on a template for AsciiDoc books.

I'm assuming you are referring to https://github.com/asciidoctor/asciidoctor.js . Rest assured that it is officially supported by @mojavelinux and the Asciidoctor team.

In my circles, Markdown is popular. But my writer friends quickly hit limitations with Markdown when it comes to creating more advanced documentation. In particular, tables are a real pain in Markdown compared to AsciiDoc. 

Tooling for AsciiDoc is also excellent. The real advantage with AsciiDoc is it's ability to output to multiple formats, by design. Particularly when we talk about l10n and i18n activities, which often rely on XML. Specifically DocBook 5.

**Comment by mojavelinux** [comment]
>  there's no official JS implementation

Asciidoctor.js is the official JS implementation of AsciiDoc. I think that by transpiling, it's actually more accurate than a clean room solution would be. It has an integration layer to close some of the gaps across different JS runtimes and we're also considering adding a simplified API.

> it might be hard for us to provide WYSIWYG features for AsciiDoc.

I'd be happy to hear what you'd need. We want it to be well suited for creating tooling.

**Comment by jaredmorgs** [comment]
Let me know if I can help further with the AsciiDoc templating stuff.
AsciiDoc support is a key reason why I'm using, and advocating for the
service.

On Wed, 15 Jun 2016 at 12:12 Dan Allen <notifications@github.com> wrote:

> there's no official JS implementation
>
> Asciidoctor.js is the official JS implementation of AsciiDoc. I think that
> by transpiling, it's actually more accurate than a clean room solution
> would be. It has an integration layer to close some of the gaps across
> different JS runtimes and we're also considering adding a simplified API.
>
> it might be hard for us to provide WYSIWYG features for AsciiDoc.
>
> I'd be happy to hear what you'd need. We want it to be well suited for
> creating tooling.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/GitbookIO/feedback/issues/31#issuecomment-226069742>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA6J2TaFQzjpNiPIQsI-_d3A4rvhf_1Lks5qL1-hgaJpZM4Iz7qp>
> .
>
-- 

Sent from Mobile.

**Comment by jaredmorgs** [comment]
I'm finding that the EPPO approach is very well suited to the "theme-faq"
template. There is some manual curation required of links, but it is quite
possible to set up a good EPPO experience with GitBook.

On Wed, 6 Jul 2016 at 22:21 Jared Morgan <jaredleonmorgan@gmail.com> wrote:

> Let me know if I can help further with the AsciiDoc templating stuff.
> AsciiDoc support is a key reason why I'm using, and advocating for the
> service.
>
> On Wed, 15 Jun 2016 at 12:12 Dan Allen <notifications@github.com> wrote:
>
>> there's no official JS implementation
>>
>> Asciidoctor.js is the official JS implementation of AsciiDoc. I think
>> that by transpiling, it's actually more accurate than a clean room solution
>> would be. It has an integration layer to close some of the gaps across
>> different JS runtimes and we're also considering adding a simplified API.
>>
>> it might be hard for us to provide WYSIWYG features for AsciiDoc.
>>
>> I'd be happy to hear what you'd need. We want it to be well suited for
>> creating tooling.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/GitbookIO/feedback/issues/31#issuecomment-226069742>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/AA6J2TaFQzjpNiPIQsI-_d3A4rvhf_1Lks5qL1-hgaJpZM4Iz7qp>
>> .
>>
> --
>
> Sent from Mobile.
>
-- 

Sent from Mobile.
